pkg/charts/charts.go

```diff
@@ -175,3 +175,41 @@ func InsertVersionValues(values chartutil.Values) (chartutil.Values, error) {
 	}
 	return chartutil.ReadValues(InsertVersion([]byte(raw)))
 }
+
+// OverrideFromFile overrides the given map with the given file from FS
+func OverrideFromFile(values map[string]interface{}, fs http.FileSystem, chartName, name string) (map[string]interface{}, error) {
+	// Load Values file
+	valuesOverride := loader.BufferedFile{
+		Name: "values-ha.yaml",
+	}
+	if err := ReadFile(fs, chartName+"/", &valuesOverride); err != nil {
+		return nil, err
+	}
+
+	var valuesOverrideMap map[string]interface{}
+	err := yaml.Unmarshal(valuesOverride.Data, &valuesOverrideMap)
+	if err != nil {
+		return nil, err
+	}
+	return MergeMaps(valuesOverrideMap, values), nil
+}
+
+// MergeMaps returns the merge result of two maps
+func MergeMaps(a, b map[string]interface{}) map[string]interface{} {
```

Comment:
we use mergo elsewhere in the project.  is that usable here?

Reply:
My reasons for choosing this over mergo:
- Mergo does not work semantically correct with bools. See https://github.com/imdario/mergo/issues/24 (obv due to another go reason)
- We had this same function previously too, but then this wasn't needed after the chart rendering with helm v3 refactor. This is also needed at other places too like [for merge during rendering](https://github.com/linkerd/linkerd2/pull/5433/files#diff-25fa6400f74d9645bcb61a83df4efb9f15d3da78281489c30fcff927cc62cb1dR179). so having a re-usable `mergeMap` which is semantically correct made sense.